bin/auth_ejabberd.php

```diff
@@ -45,43 +45,13 @@
 }
 
 use Dice\Dice;
-use Friendica\App\Mode;
-use Friendica\Core\Logger\Capability\LogChannel;
-use Friendica\Security\ExAuth;
-use Psr\Log\LoggerInterface;
 
-if (sizeof($_SERVER["argv"]) == 0) {
-	die();
-}
-
-$directory = dirname($_SERVER["argv"][0]);
-
-if (substr($directory, 0, 1) != DIRECTORY_SEPARATOR) {
-	$directory = $_SERVER["PWD"] . DIRECTORY_SEPARATOR . $directory;
-}
-
-$directory = realpath($directory . DIRECTORY_SEPARATOR . "..");
+chdir(dirname(__FILE__, 2));
```

Comment:
`dirname(__FILE__, 2)` is used 3 times in this file, wouldn't it be a good idea to make it into a variable?

Reply:
I'd say no. I would like to keep the global namespace a clean a possible. And the Dice rules file will be refactored in the future again. Personally I think keeping the hardcoded `dirname(__FILE__, 2)` in the entry points increases the readability. But we can replace it with `dirname(__DIR__, 1)`. What do you think?

Reply:
Personally, I prefer using `__DIR__` if it's used directly, meaning as a replacement for `dirname(__FILE__)` specifically. If we need to go up more folder levels, then I prefer using `dirname(__FILE__, 2+)`, I feel it is more intuitive file-tree wise.